server/server.go

```diff
@@ -171,3 +176,55 @@ func (server *HTTPServer) Start() error {
 func (server *HTTPServer) Stop(ctx context.Context) error {
 	return server.Serv.Shutdown(ctx)
 }
+
+// genericHandle
+func (server HTTPServer) genericAggregatorRedirect(writer http.ResponseWriter, request *http.Request) {
+	endpointComps := strings.Split(request.RequestURI, server.Config.APIPrefix)
+	endpoint := strings.Join(endpointComps[1:], ",")
+	endpointURL, err := url.Parse(server.ServicesConfig.AggregatorBaseEndpoint + endpoint)
+
+	if err != nil {
+		log.Error().Err(err).Msg("Error during endpoint URL parsing")
+		handleServerError(writer, err)
+	}
+
+	// test service available
+	_, err = http.Get(endpointURL.String())
+	if err != nil {
+		log.Error().Err(err).Msg("Aggregator service unavailable")
+
+		if _, ok := err.(*url.Error); ok {
+			err = &AggregatorServiceUnavailableError{}
+		}
+
+		handleServerError(writer, err)
+	}
+
+	http.Redirect(writer, request, endpointURL.String(), 302)
+}
+
+// genericHandle
+func (server HTTPServer) genericContentServiceRedirect(writer http.ResponseWriter, request *http.Request) {
```

Comment:
btw, genericAggregatorRedirect, and genericContentServiceRedirect are almost the same function, so maybe it could be something like

```golang
func (server HTTPServer) redirectTo(baseURL string) func(http.ResponseWriter, *http.Request)
```

Reply:
Done